lte/cloud/go/services/nprobe/encoding/record.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package encoding
+
+import (
+	"encoding/asn1"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"magma/lte/cloud/go/services/nprobe/obsidian/models"
+	eventdM "magma/orc8r/cloud/go/services/eventd/obsidian/models"
+
+	"github.com/gofrs/uuid"
+	"github.com/golang/glog"
+)
+
+// EpsIRIRecord represents a full IRI record combining header and payload
+type EpsIRIRecord struct {
+	Header  EpsIRIHeader
+	Payload EpsIRIContent
+}
+
+// Encode returns a byte sequence of the EpsIRIRecord in network byte order
+func (r *EpsIRIRecord) Encode() ([]byte, error) {
+	recordType := getRecordType(r.Payload.EPSEvent)
+	content, err := asn1.MarshalWithParams(r.Payload, recordType)
+	if err != nil {
+		glog.Errorf("Failed to marshal record %v\n", r)
+		return []byte{}, err
+	}
+
+	// Update payload length before marshaling the header
+	r.Header.PayloadLength = uint32(len(content))
+	msg := r.Header.Marshal()
+	return append(msg, content...), nil
+}
+
+// Decode constructs an IRI record from a byte sequence
+func (r *EpsIRIRecord) Decode(b []byte) error {
+	if len(b) < int(HeaderFixLen) {
+		return errors.New("input too small")
+	}
+
+	hdr_len := binary.BigEndian.Uint32(b[4:8])
+	pld_len := binary.BigEndian.Uint32(b[8:12])
+	if int(hdr_len+pld_len) > len(b) {
+		return errors.New("invalid input size")
+	}
+
+	if err := r.Header.Unmarshal(b[:hdr_len]); err != nil {
+		glog.Errorf("Failed to unmarshal record header %s\n", err)
```

Comment:
Nit: Generally we want to log the err as high up in the callstack as possible.

Reply:
fixed